src/sage/matroids/oriented_matroids/signed_subset_element.py

```diff
@@ -0,0 +1,850 @@
+            sage: M = OrientedMatroid(C, key='covector')
+            sage: E = M.elements()[0]
+            sage: E(1)
+            0
+            sage: E(0)
+            1
+            sage: E(2)
+            -1
+        """
+        if var in self.positives():
+            return 1
+        if var in self.negatives():
+            return -1
+        if var in self.zeros():
+            return 0
+        raise ValueError("not in groundset")
+
+    def __hash__(self):
+        """
+        Return hashed string of signed subset.
+
+        EXAMPLES::
+
+            sage: from sage.matroids.oriented_matroids.oriented_matroid import OrientedMatroid
+            sage: C = [[1, 0, -1], [-1, 0, 1],[0, 0, 0]]
+            sage: M = OrientedMatroid(C, key='covector')
+            sage: E = M.elements()[0]
+            sage: hash(E)
+            1231571227600608182
+        """
+        fsp = frozenset(self._p)
+        fsn = frozenset(self._n)
+        fsz = frozenset(self._z)
+        return hash((fsp, fsn, fsz))
+
+    def __neg__(self):
+        """
+        Return the opposite signed subset.
+
+        EXAMPLES::
+
+            sage: from sage.matroids.oriented_matroids.oriented_matroid import OrientedMatroid
+            sage: C = [[1, 0, -1], [-1, 0, 1],[0, 0, 0]]
+            sage: M = OrientedMatroid(C, key='covector')
+            sage: E = M.elements()[0]; E
+            +: 0
+            -: 2
+            0: 1
+            sage: -E
+            +: 2
+            -: 0
+            0: 1
+        """
+        N = copy.copy(self)
+        N._p = self._n
+        N._n = self._p
+        return N
+
+    def __eq__(self, other):
+        """
+        Return whether two elements are equal.
+
+        EXAMPLES::
+
+            sage: from sage.matroids.oriented_matroids.oriented_matroid import OrientedMatroid
+            sage: C = [[1, 0, -1], [-1, 0, 1],[0, 0, 0]]
+            sage: M = OrientedMatroid(C, key='covector')
+            sage: E1 = M.elements()[0]
+            sage: E2 = M.elements()[1]
+            sage: E1 == E2
+            False
+            sage: E1 == E1
+            True
+        """
+        if isinstance(other, SignedSubsetElement):
+            if self._p == other._p \
+                    and self._n == other._n \
+                    and self._z == other._z:
+                return True
+        return False
+
+    def __ne__(self, other):
+        """
+        Return whether two elements are not equal.
+
+        EXAMPLES::
+
+            sage: from sage.matroids.oriented_matroids.oriented_matroid import OrientedMatroid
+            sage: C = [[1, 0, -1], [-1, 0, 1],[0, 0, 0]]
+            sage: M = OrientedMatroid(C, key='covector')
+            sage: E1 = M.elements()[0]
+            sage: E2 = M.elements()[1]
+            sage: E1 != E2
+            True
+            sage: E1 != E1
+            False
+        """
+        return not (self == other)
+
+    def _cmp_(self, other):
```

Comment:
Remove this. This is not meant to be used. (You've also created a situation where `x < y` and `y < x`!) Implement either `_richcmp_` or `__lt__`. I would recommend `_richcmp_` (which might need a metaclass), which would also cover the `__eq__` and `__ne__`.

Reply:
Yes, because the `_cmp_` function was required for something, but we don't actually care about the comparison itself. (I forget now why we need it, but other problems were happening without a `_cmp_` function.) I'll see about implementing one of the other ones instead then.

Reply:
Ok, looking at the code, this was added so that partial orders can be used. Without this function we get errors when building partial orders. (The order/cmp function itself doesn't matter, but something needs to exist). So the point of this function isn't actually to compare things.

But I just removed the function and ran the tests and everything seems to be working fine without it. So maybe this was fixed by some other way and so it's no longer needed. I've gone ahead and removed it as there's no point to having this comparison function.

Reply:
Do you think it is worthwhile to add in partial order comparisons? If so, you need to tweak the metaclasses and implement a `_richcmp_` (maybe `__richcmp__`?) method.